src/main/java/me/steven/carrier/mixin/MixinPlayerEntity.java

```diff
@@ -37,4 +42,17 @@ private void b(CallbackInfoReturnable<Boolean> cir) {
             cir.setReturnValue(false);
         }
     }
+    @Inject(method = "dropInventory", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/player/PlayerInventory;dropAll()V", shift = At.Shift.AFTER))
+    private void c(CallbackInfo ci) {
+        CarrierComponent carrier = Carrier.HOLDER.get(this);
+        CarryingData carrying = carrier.getCarryingData();
+        if (carrying != null) {
+            Carriable<?> carriable = CarriableRegistry.INSTANCE.get(carrying.getType());
+            BlockPos pos = this.getBlockPos();
+            if (!world.isClient && carriable != null && world.getBlockState(pos).getMaterial().isReplaceable()) {
+                carriable.tryPlace(carrier, world, new CarriablePlacementContext(carrier, carriable, pos, Direction.DOWN, this.getHorizontalFacing()));
```

Comment:
If it fails to place the block in the position the player is in it will void what was being carried, which can be the case with gravestone/death chest mods or similar things.

Reply:
any suggested fix? (as i see it still better then teleporting the item with the player)

Reply:
i mean  if you prefer it being teleported in case it fails that's pretty easy just asking if you got any ideas as how to avoid that

Reply:
You should check in a range to see if there's any nearby blocks it can be placed. If it still fails, save to a persistent state with an UUID and print it in the log. 
Checking a range of blocks could be too slow to have it run on the dropInventory logic, I'd add the player + death pos to a queue (also persistent) and then try to find a position to place the block on the server tick

Reply:
that would work very nicely although honestly it's a tad more effort then i'm willing to put into fixing a minor bug a player on my server complained about...
did change it so no longer voids but feel free to close if that's not sufficent

Reply:
I'll try to implement this soon, thanks for the PR and reminding me of the issue.